features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/BitmapPool.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.content.ComponentCallbacks2
+import android.content.res.Configuration
+import android.graphics.Bitmap
+import android.graphics.Bitmap.Config
+import android.util.LruCache
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.sessionreplay.internal.utils.CacheUtils
+import java.util.concurrent.atomic.AtomicInteger
+
+internal object BitmapPool : Cache<String, Bitmap>, ComponentCallbacks2 {
+    @Suppress("MagicNumber")
+    private val MAX_CACHE_MEMORY_SIZE_BYTES = 4 * 1024 * 1024 // 4MB
+
+    private var bitmapsBySize = HashMap<String, HashSet<Bitmap>>()
```

Comment:
It's a problem to inject  for testing because BitmapPool is a singleton

Reply:
yup I just noticed that, I think @0xnm will challenge you on that and ask why don't you use a non - singleton instance here and just make sure you inject one single instance when you initialize the SR feature.

Reply:
yes, I agree although it's the same issue that we had with base64lrucache. To use a single instance I would have to hold that instance higher up, and then it needs to be injected into base64serializer. Base64serializer is constructed in the baseMapper so that all mappers can make use of it, without us modifying the signature of map() - but since it's constructed there we have a problem injecting dependencies.

Reply:
I see your point and it might be the easiest way. On the other hand if we would find a way to use non - singleton instances here we will be able to solve couple of things:
- get rid of those testing methods
- eliminate the problem you raised about having collisions if you use 2 datadog instances (2 SR instances sharing same singleton). 
I am not saying that you should do this here but just try to assess the effort and evaluate if it's really doable here or if you want to keep it for later.

Reply:
I'll refactor the cache objects to single class instances in the next pr